packages/block-library/src/index.js

```diff
@@ -148,3 +155,40 @@ export const registerCoreBlocks = ( enableLegacyWidgets, enableMenu ) => {
 		setGroupingBlockName( group.name );
 	}
 };
+
+/**
+ * Check conditions to register an experimental block.
+ *
+ * @param {boolean} setting    The editor setting for the block.
+ *
+ * @return {boolean} if all conditions allow block to be registered.
+ */
+const checkExperimentalBlockConditions = ( setting ) => {
+	const isSecondPhase = process.env.GUTENBERG_PHASE === 2 ? true : false;
```

Comment:
I don't think this approach will work with dead code elimination (though I'm not 100% sure it works anyway):

Some more info here.
https://github.com/WordPress/gutenberg/blob/c1235d4486b36bab451f57866dc323451dc76963/docs/designers-developers/developers/feature-flags.md

Might be easier to add an early return from `registerExperimentalCoreBlocks`:
```
export const registerExperimentalCoreBlocks = ( settings ) => {
    if ( process.env.GUTENBERG_PHASE !== 2 ) {
        return;
    }
```

Reply:
Oooh I see. But in that case shouldn't the whole experiments settings page be behind the same feature flag?

Reply:
Yep, because it's part of the plugin codebase, it should only appear when the plugin is activated.